### PR TITLE
fix: fix bug in filesystem.insertLink

### PR DESCRIPTION
**Pull request by jrainlau**
In MacOS, if I run `asar.createPackage()` to pack a folder into `.asar`, which contains "symlink", it would get wrong after I run `asar.extractAll()` from the generated `.asar`.

Let's look at this Beyond Compare result:

企业微信截图_0ef755c7-3d85-4037-8d72-906e533b1fe1

The folder in left is the origin one, and the folder in right was generated by using `asar.extractAll()` from the origin one's `.asar` ( folderLeft --> folderLeft.asar --> folderRight).

Inside the left folder, file **Resources** were pointed to `Versions/Current/Resources`, but the same file inside the right folder were pointed to `Versions/A/Resources`, it's wrong.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@
 .node-version
 npm-debug.log
 .idea
+.vscode
```
Comment (ckerr) [inline .gitignore:11]: This change seems to be unrelated to the fix

Comment (jrainlau) [reply .gitignore]: Sorry for my mistake. I just removed it in 07a14a8
